### PR TITLE
Reinforcements chance to be real

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:

- Gives the fn_reinforceSide.sqf a chance, decreasing with number of units, to spawn in real reinforcements regardless of distance from players
- adjusted fn_patrolReinf.sqf to use a more varied vehicle set

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Comment by ante185** [comment]
updated probability equation to exclude greenfor units. the random 120 may still be too low though?
As there isn't any filtering or bias for distance / being closer to players it's still gonna be fairly unlikely to run into these reinforcements at moderate to low unit counts

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf [hunk 1]
```diff
@@ -15,11 +15,17 @@ ServerInfo_5("Spawning PatrolReinf. Dest:%1 Orig:%2 Size:%3 Side:%4 Land:%5",_mr
 
 private _vehicleType = if (_isLand) then {
 	[_mrkOrigin, 1] call A3A_fnc_addTimeForIdle;
-	selectRandom (_faction get "vehiclesTrucks");
+	private _transportTruck = selectRandom ((_faction get "vehiclesTrucks") + (_faction get "vehiclesLightArmedTroop"));
+	if (count _groupType <= 4) then { _transportTruck = selectRandom (_faction get "vehiclesMilitiaCars") } //Militia cars has the capacity, military usual includes Hunter/M-ATV and similar with only 3 cargo seats
```
Comment (jaj22) [inline A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf:19]: Maybe a danger here that some shitty WW2 cars don't have four seats.

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf [hunk 2]
```diff
@@ -15,11 +15,17 @@ ServerInfo_5("Spawning PatrolReinf. Dest:%1 Orig:%2 Size:%3 Side:%4 Land:%5",_mr
...
+	else
+	{
+		if (round _quality < 3) then { _transportTruck = selectRandom (_faction get "vehiclesMilitiaTrucks") } //low quality gets the militia truck
+	};
+	_transportTruck;
 } else {
 	private _transportPlanes = _faction get "vehiclesPlanesTransport";
 	private _transportHelis = _faction get "vehiclesHelisTransport";
-	if (count _groupType <= 4) then { _transportHelis append (_faction get "vehiclesHelisLight") };
+	if (count _groupType <= 4 and count _faction get "vehiclesHelisLight" > 0) then { _transportHelis = (_faction get "vehiclesHelisLight") };
```
Comment (jaj22) [inline A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf]: Similarly I don't think light helis are guaranteed to have four cargo seats? Although we do have a separate patrol category so maybe they should.

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf [hunk 3]
```diff
@@ -15,11 +15,17 @@ ServerInfo_5("Spawning PatrolReinf. Dest:%1 Orig:%2 Size:%3 Side:%4 Land:%5",_mr
 
 private _vehicleType = if (_isLand) then {
 	[_mrkOrigin, 1] call A3A_fnc_addTimeForIdle;
-	selectRandom (_faction get "vehiclesTrucks");
+	private _transportTruck = selectRandom ((_faction get "vehiclesTrucks") + (_faction get "vehiclesLightArmedTroop"));
+	if (count _groupType <= 4) then { _transportTruck = selectRandom (_faction get "vehiclesMilitiaCars") } //Militia cars has the capacity, military usual includes Hunter/M-ATV and similar with only 3 cargo seats
+	else
+	{
+		if (round _quality < 3) then { _transportTruck = selectRandom (_faction get "vehiclesMilitiaTrucks") } //low quality gets the militia truck
```
Comment (jaj22) [inline A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf]: Should probably be <2? 2 is regulars, 3 is specops.

File: A3A/addons/core/functions/CREATE/fn_reinforceSide.sqf [hunk 4]
```diff
@@ -137,7 +137,7 @@ while {_totalReinf > 0} do
         [_marker, _numTroops, _quality] remoteExecCall ["A3A_fnc_garrisonServer_addUnitCount", 2];
         continue;
     };
-    if (_rebelsNear or _rebelSpawners inAreaArray [markerPos _source, 1000, 1000] isNotEqualTo []) then {
+    if (_rebelsNear or _rebelSpawners inAreaArray [markerPos _source, 1000, 1000] isNotEqualTo [] or ((count (units independent))+(random 120) > ((count allUnits)))) then {
```
Comment (jaj22) [inline A3A/addons/core/functions/CREATE/fn_reinforceSide.sqf]: Brackets are a bit extreme here. Also clearer to do count occupants + count invaders instead of this count allUnits - count independent. Civ counts are controlled separately anyway.

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf (hunk 3 above)
Comment (ante185) [reply A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf]: oh i see what happened, i read L10-L11 wrong, I missed that the Xsquads and Xmedium were combined

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf (hunk 2 above)
Comment (ante185) [reply A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf]: light helis should absolutely have at least 4 seats, if they don't then it's the template that's wrong

File: A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf (hunk 1 above)
Comment (ante185) [reply A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf:19]: that is true, it should trim the AI that can't fit in the vehicle though so are there any other problems in the situation?
Otherwise I can just revert to using trucks

Comment (jaj22) [reply A3A/addons/core/functions/CREATE/fn_patrolReinf.sqf:19]: Actually this function is written better than I remembered. Might work more generally than it's used. Test a few corner cases if you can.

File: A3A/addons/core/functions/CREATE/fn_reinforceSide.sqf (hunk 4 above)
Comment (ante185) [reply A3A/addons/core/functions/CREATE/fn_reinforceSide.sqf]: would it be much of a difference? I can make that change

Also, 120 is kinda pulled outta thin air and looking at about how many AI are about when only a few locations are spawned, and suggestion on math adjustments?

Comment (jaj22) [reply A3A/addons/core/functions/CREATE/fn_reinforceSide.sqf]: I feel like ideally you'd use a higher number for more players, because that implies they're playing on a server that can handle it. But as I intentionally deleted most of those mechanisms I'm not *super* happy about the idea.

Should have an extra condition for whether the target is within mission distance, so that it's not generating these on the other side of the map after an HQ move. So the full condition would be:

`if (_rebelsNear or _rebelSpawners inAreaArray [markerPos _source, 1000, 1000] isNotEqualTo []
or (markerPos _marker distance2d markerPos "Synd_HQ" < distanceMission and count units Invaders + count units Occupants < random 120))`

(Untested)

120 seems... ok to me. Would probably go a bit higher because it's average 60, but then it might be hitting localhosts a bit hard. I suppose fewer units enemies die with fewer players though.

**Review by jaj22 — APPROVED** [review]
